lib/src/ovis_util/util.h

```diff
@@ -107,17 +109,54 @@ typedef struct string_ref_s {
 	LIST_ENTRY(string_ref_s) entry;
 } *string_ref_t;
 
-struct attr_value_list {
+typedef struct attr_value_list {
 	int size; /* capacity of list */
 	int count; /* number of keys in the list */
 	LIST_HEAD(string_list, string_ref_s) strings;
 	struct attr_value list[OVIS_FLEX];
-};
+} *av_list_t;
+
+/** \brief Get the value of an attribute in an attr-value list
+ *
+ * Search the \c avl for the specified \c name. If found, return the string value
+ * after applying ${} substitions.
+ *
+ * \param av_list The attribute-value list
+ * \param name The attribute name
+ *
+ * \returns The attribute name with ${} substitution or NULL if \c name is not found.
+ */
+char *av_value(struct attr_value_list *av_list, const char *name);
 
 /**
  * \brief Get the value of attribute \c name
+ *
+ * Search the \c avl for the specified \c name. If found, return the string value
+ * after replacing environement variables  with their values. If not found, perform
+ * the same operation only using the value specified by \c default.
+ *
+ * \param av_list The attribute-value list
+ * \param name The attribute name
+ * \param def The default value to use if \c name is not found.
+ *
+ * \returns The attribute name with ${} substitution or NULL if there is a
+ *          memory allocation failure.
  */
-char *av_value(struct attr_value_list *av_list, const char *name);
+char *av_value_default(struct attr_value_list *av_list, const char *name, const char *def);
+
+/**
+ * \brief Return an AVL given string containing name,value pairs
+ *
+ * Separates \c input into <name>=<value> pairs. The separater for
+ * splitting the input string is specified by \c sep.
+ *
+ * \param input A NULL terminated string containing <name>=<value> pairs
+ * \param sep A string specifying the input string separator character.
+ *
+ * \returns An avl_list_t containing name,value pairs or NULL if there
+ * is a memory allocation error.
+ */
+av_list_t av_value_list(const char *input, const char *sep);
```

Comment:
Should `sep` be `char`? It seems that the implementation at line `ovis.c:397` assumes that `sep` is a single character. What do you think?